### PR TITLE
Remove md array: it is not used any more.

### PR DESCRIPTION
**Pull request by snhenson**
##### Checklist

- [x] tests are added or updated
##### Description of change

This removes the md array. The changes to signature algorithm handling mean it is no longer used.

### CODE REVIEW

**Comment by dot-asm** [comment]
Could you comment on red cross from travis? The one with "boring" tests of course...

**Comment by snhenson** [comment]
Checking it. It should be possible to reproduce it with our own signature algorithms tests.

**Comment by snhenson** [comment]
Finally got this one to pass all the tests. It ended up a bit messier than I hoped but a later tidy up/overhaul of tls1_check_chain() should improve things a lot.

File: ssl/statem/statem_clnt.c [hunk 1]
```diff
@@ -2240,11 +2240,9 @@ MSG_PROCESS_RETURN tls_process_certificate_request(SSL *s, PACKET *pkt)
             goto err;
         }
 
-        /* Clear certificate digests and validity flags */
-        for (i = 0; i < SSL_PKEY_NUM; i++) {
-            s->s3->tmp.md[i] = NULL;
+        /* Clear certificate validity flags */
+        for (i = 0; i < SSL_PKEY_NUM; i++)
             s->s3->tmp.valid_flags[i] = 0;
```
Comment (richsalz) [inline ssl/statem/statem_clnt.c]: Every time I see this kind of thing, I weep. :)
```
{ declare uint32_t *fp; }
for (fp = s->s3->tmp.valid_flags, i = SSL_PKEY_NUM; --i >= 0; fp++)
    *fp = 0;
```

Back when things like this counted ... :)

File: ssl/t1_lib.c [hunk 2]
```diff
@@ -1101,30 +1072,45 @@ int tls1_set_server_sigalgs(SSL *s)
...
+    for (i = 0; i < SSL_PKEY_NUM; i++)
         s->s3->tmp.valid_flags[i] = 0;
-    }
+    /*
+     * If peer sent no signature algorithms check to see if we support
+     * the default algorithm for each certificate type
+     */
+    if (s->s3->tmp.peer_sigalgs == NULL) {
+        const uint16_t *sent_sigs;
+        size_t sent_sigslen;
```
Comment (richsalz) [inline ssl/t1_lib.c]: merge these two lines?

File: ssl/statem/statem_clnt.c (hunk 1 above)
Comment (snhenson) [inline ssl/statem/statem_clnt.c]: I used to do that a lot. Compilers can presumably optimise that kind of thing automaticaly these days.

At some point valid_flags is going to go away anyway.

Comment (richsalz) [inline ssl/statem/statem_clnt.c]: WFM. :)

File: ssl/t1_lib.c (hunk 2 above)
Comment (snhenson) [inline ssl/t1_lib.c]: Done.

**Comment by richsalz** [comment]
+1

**Pull request closed without merging**